glean-core/src/lib.rs

```diff
@@ -308,22 +334,27 @@ impl Glean {
         self.upload_enabled
     }
 
-    /// Handles the changing of state when upload_enabled changes.
+    /// Handles the changing of state from upload disabled to enabled.
     ///
     /// Should only be called when the state actually changes.
-    /// When disabling, all pending metrics, events and queued pings are cleared.
     ///
-    /// When enabling, the core Glean metrics are recreated.
+    /// The core Glean metrics are recreated.
```

Comment:
the docs should call out that the flag is not yet set to `true` when this message should be called.

Reply:
But that's not strictly true -- it might be set to true already if this is called during initialization.

Reply:
Ah right.
But it is strictly true that afterwards it's definitely set to true.

Reply:
Got it.  Adding this to the comment (and `on_upload_disabled`, too)